packages/ui/src/AppSidebar/AppSidebar.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ReactNode, useCallback, useState } from 'react'
+import cn from 'classnames'
+import { ChevronsLeft, ChevronsRight } from 'react-feather'
+import createPersistedState from 'use-persisted-state'
+
+import { IconButton } from '../IconButton'
+import { useOpenCloseState } from '../hooks'
+import { AppSidebarFavorites } from './AppSidebarFavorites'
+import { appSidebarContext, FavoriteItem } from './appSidebarContext'
+
+import styles from './AppSidebar.module.scss'
+
+export interface AppSidebarProps {
+  children?: ReactNode
+  className?: string
+}
+
+export const AppSidebar = ({ children, className }: AppSidebarProps) => {
+  const { isOpen, toggle } = useOpenCloseState(true)
+  const usePersistedFavoritesState = createPersistedState<string[]>('sideBarFavorites')
+  const [favorites, setFavorites] = usePersistedFavoritesState([])
+  const [registeredFavorites, setRegisteredFavorites] = useState<Record<string, FavoriteItem>>({})
+
+  const registerFavorite = useCallback(
```

Comment:
I didn't get the idea of the whole `register` stuff besides `favorites` 😅

Reply:
Favorites section appears autocratically if you have items in expandable sections